src/axom/slam/examples/HandleMesh.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
```

Comment:
New Slam example where the `PositionType` is an integer, but the `ElementType` is a "handle" type that wraps an integer.

Reply:
Just so I understand `slam::PositionType` is a default type, that is an integer?

Reply:
Yes.  They are defined here:
https://github.com/LLNL/axom/blob/6704667d9981e6d3cf9aaffb8c92a560f7809a1a/src/axom/slam/Utilities.hpp#L22-L24

Reply:
Thanks. I was a little confused b/c use of PosType throughout the implementation of the templates and slam::PositionType as a concrete type. 

For example, cases like:
PosType = slam::PositionType

It might help to call these things like:

`slam::DefaultPositionType`

Reply:
Great suggestion @cyrush -- I'll try to make that change.

Reply:
Done. I think this is a lot cleaner.
...and it also uncovered a few places where the code was using the wrong type!

Thanks @cyrush.